pkg/cli/image/manifest/manifest.go

```diff
@@ -293,8 +293,34 @@ func FirstManifest(ctx context.Context, from imagereference.DockerImageReference
 	return srcManifest, ManifestLocation{Manifest: srcDigest}, nil
 }
 
-// ManifestToImageConfig takes an image manifest and converts it into a structured object.
-func ManifestToImageConfig(ctx context.Context, srcManifest distribution.Manifest, blobs distribution.BlobService, location ManifestLocation) (*dockerv1client.DockerImageConfig, []distribution.Descriptor, error) {
+// ManifestToImageConfig takes an image manifest and converts it into a DockerImageConfig and digest Descriptors.
+// If srcManifest is a DeserializedManifestList, the platformSpec determines which manifest (os/arch) will be used.
+func ManifestToImageConfig(ctx context.Context, srcManifest distribution.Manifest, blobs distribution.BlobService, manifests distribution.ManifestService, location ManifestLocation, platformSpec string) (*dockerv1client.DockerImageConfig, []distribution.Descriptor, error) {
+	switch t := srcManifest.(type) {
+	case *manifestlist.DeserializedManifestList:
+		matched := false
+		for _, manifest := range t.Manifests {
+			childManifest, err := manifests.Get(ctx, manifest.Digest, distribution.WithManifestMediaTypes([]string{manifestlist.MediaTypeManifestList, schema2.MediaTypeManifest}))
```

Comment:
`manifests.Get` is a relatively expensive operation. The result will be thrown away if `manifest.Platform` doesn't match platformSpec.

Reply:
updated, moved this all to FirstManifest

Reply:
The manifest `childManifest` is still downloaded before checking that it is needed.

Reply:
oh right, thanks, updated, ptal